stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -40,6 +41,30 @@ julia> diagm([7, 13])
 2×2 Matrix{$Int}:
  7   0
  0  13
+
+julia> ans + I
+2×2 Matrix{Int64}:
+ 8   0
+ 0  14
+
+julia> I(2)
+2×2 Diagonal{Bool, Vector{Bool}}:
+ 1  ⋅
+ ⋅  1
+```
+
+Note that a one-column matrix is not treated like a vector, but instead calls the
+method `Diagonal(A::AbstractMatrix)` which extracts 1-element `diag(A)`:
```

Comment:
Not sure if it would work as suggested, but should we cross-reference here?

```suggestion
method [`Diagonal(A::AbstractMatrix)`](@ref) which extracts 1-element `diag(A)`:
```

Reply:
If this did work, when would it be useful? At the repl I think you can't follow links, and on the website the two are adjacent:

https://docs.julialang.org/en/v1/stdlib/LinearAlgebra/#LinearAlgebra.Diagonal